exporter/src/main/java/io/zeebe/exporters/kafka/producer/DefaultKafkaProducerFactory.java

```diff
@@ -73,7 +74,18 @@
     options.putAll(config.getProducer().getConfig());
 
     options.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, RecordIdSerializer.class);
-    options.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, RecordSerializer.class);
+    switch (config.getProducer().getFormat()) {
+      case JSON:
+        {
+          options.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, RecordSerializer.class);
+          break;
+        }
+      case PROTOBUF:
+        {
+          options.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, ProtobufRecordSerializer.class);
+          break;
+        }
```

Comment:
Can you add a `default` branch which throws an `IllegalArgumentException` with an appropriate message (e.g. `String.format("Expected format to be one of JSON or PROTOBUF, but got %s", config.getProducer().getFormat())`)? It's good practice with enums in general in case they get modified and we forget to update a place where we switch over them :slightly_smiling_face: